src/platform/windows/display_uwp.cpp

```diff
@@ -0,0 +1,247 @@
+  using namespace Windows::Foundation;
+  using namespace Windows::Graphics::Capture;
+  using namespace Windows::Graphics::DirectX::Direct3D11;
+
+  extern "C" {
+  HRESULT __stdcall CreateDirect3D11DeviceFromDXGIDevice(::IDXGIDevice *dxgiDevice, ::IInspectable **graphicsDevice);
+  }
+  struct
+#if WINRT_IMPL_HAS_DECLSPEC_UUID
+    __declspec(uuid("A9B3D012-3DF2-4EE3-B8D1-8695F457D3C1"))
+#endif
+      IDirect3DDxgiInterfaceAccess: ::IUnknown {
+    virtual HRESULT __stdcall GetInterface(REFIID id, void **object) = 0;
+  };
+}  // namespace winrt
+#if !WINRT_IMPL_HAS_DECLSPEC_UUID
+static constexpr GUID GUID__IDirect3DDxgiInterfaceAccess = {
+  0xA9B3D012, 0x3DF2, 0x4EE3, { 0xB8, 0xD1, 0x86, 0x95, 0xF4, 0x57, 0xD3, 0xC1 }
+  // compare with __declspec(uuid(...)) for the struct above.
+};
+template <>
+constexpr auto
+__mingw_uuidof<winrt::IDirect3DDxgiInterfaceAccess>() -> GUID const & {
+  return GUID__IDirect3DDxgiInterfaceAccess;
+}
+#endif
+
+namespace platf::dxgi {
+  static HRESULT
+  create_item_for_monitor(HMONITOR mon, winrt::GraphicsCaptureItem &item) {
+    auto interop_factory = winrt::get_activation_factory<winrt::GraphicsCaptureItem, IGraphicsCaptureItemInterop>();
+    return interop_factory->CreateForMonitor(mon, winrt::guid_of<winrt::Windows::Graphics::Capture::IGraphicsCaptureItem>(), winrt::put_abi(item));
+  }
+
+  uwp_capture_t::uwp_capture_t() {
+    InitializeCriticalSection(&frame_lock);
+    InitializeConditionVariable(&frame_present_cv);
+  }
+
+  uwp_capture_t::~uwp_capture_t() {
+    capture_session.Close();
+    frame_pool.Close();
+    item = nullptr;
+    capture_session = nullptr;
+    frame_pool = nullptr;
+  }
+
+  int
+  uwp_capture_t::init(display_base_t *display, const ::video::config_t &config) {
+    HRESULT status;
+    dxgi::dxgi_t dxgi;
+    winrt::com_ptr<::IInspectable> d3d_comhandle;
+    if (!winrt::GraphicsCaptureSession::IsSupported()) {
+      BOOST_LOG(error) << "Screen capture is not supported on this device for this release of Windows!"sv;
+      return -1;
+    }
+    if (FAILED(status = display->device->QueryInterface(IID_IDXGIDevice, (void **) &dxgi))) {
+      BOOST_LOG(error) << "Failed to query DXGI interface from device [0x"sv << util::hex(status).to_string_view() << ']';
+      return -1;
+    }
+    if (FAILED(status = winrt::CreateDirect3D11DeviceFromDXGIDevice(*&dxgi, d3d_comhandle.put()))) {
+      BOOST_LOG(error) << "Failed to query WinRT DirectX interface from device [0x"sv << util::hex(status).to_string_view() << ']';
+      return -1;
+    }
+
+    uwp_device = d3d_comhandle.as<winrt::IDirect3DDevice>();
+    DXGI_OUTPUT_DESC output_desc;
+    display->output->GetDesc(&output_desc);
+    if (FAILED(status = create_item_for_monitor(output_desc.Monitor, item))) {
+      BOOST_LOG(error) << "Failed to activate GraphicsCaptureItem for monitor [0x"sv << util::hex(status).to_string_view() << ']';
+      return -1;
+    }
+    if (config.dynamicRange && display->is_hdr())

```

Comment:
We shouldn't check `display->is_hdr()` because it's possible that the display is not in HDR mode yet.

We should only check `config.dynamicRange`, though even that isn't totally correct because of multiple clients with different HDR capabilities. It's probably good enough though.